library/src/main/java/com/datatorrent/lib/state/managed/ManagedStateImpl.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.lib.state.managed;
+
+import java.util.concurrent.Future;
+
+import javax.validation.constraints.Min;
+
+import com.google.common.util.concurrent.Futures;
+
+import com.datatorrent.lib.state.BucketedState;
+import com.datatorrent.netlet.util.Slice;
+
+/**
+ * Basic implementation of {@link ManagedState} where window is used to sub-group key of a particular bucket.<br/>
+ *
+ */
+public class ManagedStateImpl extends AbstractManagedStateImpl implements BucketedState
+{
+
+  public ManagedStateImpl()
+  {
+    this.numBuckets = 1;
+  }
+
+  @Override
+  public void put(long bucketId, Slice key, Slice value)
+  {
+    long timeBucket = timeBucketAssigner.getTimeBucketFor(windowId);
```

Comment:
@ilooner suggested to keep initial time and add millis corresponding to app window to it. So made that change.